### PR TITLE
Workspace name, prebuilt devcontainer, livereload

### PR DESCRIPTION
**Pull request by RossBugginsNHS**
Including:

- Renamed workspace to be project. Utilised settings for windows.title to set to be nice name
- Updated to used pre built dev container
- Removed livereload on jekyll, speeds up dev container version (F5 still works, no need to stop debugging)
- Remove sonar properties file, confirmed they are set in action variables.
- Added rubylsp gem for debug
- Added custom version number generation logic.
- Use make build in deploy pipeline.
- Added file excludes to the .vscode settings in the website folder to hide node, bundle etc from view
- Added "core" other folders as workspace folders
- Exclude package.json and package lock from jekyll built output  
- Corrected jekyll config to not have 2 plugins sections
- Upgraded pa11y versions
- Upgraded ruby version in workflow

The dev container gets ready now in under a minute, and pages quick to respond.

Only thing I haven't tested in the deploy pipeline using the make build.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

### CODE REVIEW

File: project.code-workspace [hunk 1]
```diff
@@ -80,10 +38,81 @@
...
+      "pythonist",
+      "rossbugginsnhs",
+      "ruleset",
+      "Stefaniuk"
+    ],
+    "editor.formatOnSave": true,
+    "editor.wordWrap": "off",
+
+    "githubCodeOwners.format.enabled": true,
+    "markdownlint.config": {
```
Comment (bhansell1) [inline project.code-workspace:106]: nice! Does this overwrite the markdown config file?

**Comment by RossBugginsNHS** [comment]
@bhansell1 could you test out creating a codespace from this branch please and check F5 etc to make sure the served remote pages run as well for you as seems to for me please? Cheers, Ross

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -56,44 +41,22 @@
...
-      "upgradePackages": true,
-      "username": "automatic",
-      "userUid": "automatic",
-      "userGid": "automatic"
-    }
-  }
-  // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
-  // "remoteUser": "root"
+  "forwardPorts": [4000],
+  "image": "ghcr.io/nhsdigital/nhs-notify-template-repository:latest",
```
Comment (bhansell1) [inline .devcontainer/devcontainer.json:59]: This image needs updating with the installs in the Dockerfile. The accessibility tests can't run locally. 

Unsure if they still run in the pipeline.

Comment (bhansell1) [reply .devcontainer/devcontainer.json:59]: Seems fine when running in CI. Perhaps we just have this run in CI and ignore trying to run it locally?

**Comment by bhansell1** [comment]
> @bhansell1 could you test out creating a codespace from this branch please and check F5 etc to make sure the served remote pages run as well for you as seems to for me please? Cheers, Ross

Runs when running in debug! It's also lightening fast compared to main! Nice changes!

**Review by bhansell1 — APPROVED** [review]

**Review by jakecosgrove — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (RossBugginsNHS) [reply .devcontainer/devcontainer.json:59]: Can you do a PR please to https://github.com/NHSDigital/nhs-notify-repository-template/blob/main/src/jekyll-devcontainer/src/.devcontainer/Dockerfile with required changes.

Going forward will look at creating specific images per subdomain team, but for now will keep in one as we build up a base line

**Review by bhansell1 — APPROVED** [review]

**Review by aidenvaines-bjss — APPROVED** [review]

**Pull request merged**